From a6a7b53b6312ac06df87e501bf65340b457d0aaf Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Mon, 3 Aug 2015 10:06:05 +0200
Subject: [PATCH] linux: Assume usbfs path = /dev/bus/usb when using UDEV

On some exotic hardware, e.g. HP ProLiant Moonshot Cartridge servers there
are no usb controllers, so no usb devices at all.

In this case currently libusb_init will fail, because find_usbfs_path
fails. Many apps don't handle this gracefully, and even if they do not crash
the result still is not pretty, e.g.:

unable to initialize libusb: -99

Where one simply would expect empty output.

Since on systems using udev the usbfs path should always be /dev/bus/usb
(as that gets created by udev), simply assume /dev/bus/usb when build with
USE_UDEV and the path cannot be found in the traditional way.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 libusb/os/linux_usbfs.c | 8 ++++++++
 1 files changed, 6 insertions(+)

diff --git a/libusb/os/linux_usbfs.c b/libusb/os/linux_usbfs.c
index 9d9d00e..5164616 100644
--- a/libusb/os/linux_usbfs.c
+++ b/libusb/os/linux_usbfs.c
@@ -197,6 +197,12 @@ static const char *find_usbfs_path(void)
 			ret = path;
 	}
 
+/* On udev based systems without any usb-devices /dev/bus/usb will not
+ * exist. So if we've not found anything simply assume /dev/bus/usb
+ * rather then making libusb_init fail. */
+	if (ret == NULL)
+		ret = "/dev/bus/usb";
+
 	usbi_dbg("found usbfs at %s", ret);
 	return ret;
 }
-- 
2.5.0
